APMrover2/GCS_Mavlink.cpp

```diff
@@ -818,18 +820,18 @@ void GCS_MAVLINK_Rover::handleMessage(mavlink_message_t* msg)
         RC_Channels::set_override(uint8_t(rover.rcmap.roll() - 1), roll, tnow);
         RC_Channels::set_override(uint8_t(rover.rcmap.throttle() - 1), throttle, tnow);
 
+        // a manual control message is considered to be a 'heartbeat' from the ground station for failsafe purposes
+        rover.failsafe.last_heartbeat_ms = tnow;
         break;
     }
 
     case MAVLINK_MSG_ID_HEARTBEAT:
         {
-            // We keep track of the last time we received a heartbeat from our GCS for failsafe purposes
+            // we keep track of the last time we received a heartbeat from our GCS for failsafe purposes
             if (msg->sysid != rover.g.sysid_my_gcs) {
                 break;
             }
-
-            rover.last_heartbeat_ms = AP_HAL::millis();
-            rover.failsafe_trigger(FAILSAFE_EVENT_GCS, false);
+            rover.failsafe.last_heartbeat_ms = AP_HAL::millis();
```

Comment:
Why is this setting it to AP_HAL::millis and the rest do it to tnow?

Reply:
tnow is set to millis() higher up.

Reply:
I saw that, but I think for consistency should it be set to the same as the others? Or set the others to AP_HAL::millis perhaps?